src/api.jl

```diff
@@ -429,7 +492,11 @@ function infer(fieldids::Vector{Tuple{Int, Int, Int}},
     # fitting the PSF for the whole catalog above)
     info("fitting PSF for all relevant sources")
     tic()
-    ModelInit.fit_object_psfs!(mp, target_sources, images)
+    if Threaded
+      fit_object_psfs_threaded!(mp, target_sources, images)
+    else
+      ModelInit.fit_object_psfs!(mp, target_sources, images)
+    end
     times.fit_psf = toq()
```

Comment:
This is only true for Julia 0.5, right?  To be compatible with 0.4 we'd
need to make your threading definitions from api.jl global.  That would be
OK with me if you feel strongly.  IMO, until we test with multiple threads,
there's not much point.

-- Ryan.  Sent from a phone.  Nevertheless, all typos are deliberate.
On Apr 1, 2016 8:39 PM, "kpamnany" <notifications@github.com> wrote:

> In src/api.jl
> <https://github.com/jeff-regier/Celeste.jl/pull/206#discussion_r58287864>:
>
> > @@ -429,7 +492,11 @@ function infer(fieldids::Vector{Tuple{Int, Int, Int}},
> >      # fitting the PSF for the whole catalog above)
> >      info("fitting PSF for all relevant sources")
> >      tic()
> > -    ModelInit.fit_object_psfs!(mp, target_sources, images)
> > +    if Threaded
> > +      fit_object_psfs_threaded!(mp, target_sources, images)
> > +    else
> > +      ModelInit.fit_object_psfs!(mp, target_sources, images)
> > +    end
> >      times.fit_psf = toq()
>
> It should be possible to have a single function fit_object_psfs!() rather
> than separate ones for threads and serial -- see infer(). There should be
> no need to test Threaded because if it is disabled, things like @threads
> do nothing, and threadid() and nthreads() always return 1.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/jeff-regier/Celeste.jl/pull/206/files/556cd9505c0a3614ffd7b59bee871515ae1c2982#r58287864>
>